pkg/controllers/workspace_controller.go

```diff
@@ -577,3 +661,27 @@ func (c *WorkspaceReconciler) watchMachines() handler.EventHandler {
 			}
 		})
 }
+
+// watches for nodeClaim with labels indicating workspace name.
+func (c *WorkspaceReconciler) watchNodeClaims() handler.EventHandler {
+	return handler.EnqueueRequestsFromMapFunc(
+		func(ctx context.Context, o client.Object) []reconcile.Request {
+			nodeClaimObj := o.(*v1beta1.NodeClaim)
+			name, ok := nodeClaimObj.Labels[kaitov1alpha1.LabelWorkspaceName]
+			if !ok {
+				return nil
+			}
+			namespace, ok := nodeClaimObj.Labels[kaitov1alpha1.LabelWorkspaceNamespace]
+			if !ok {
+				return nil
+			}
+			return []reconcile.Request{
+				{
+					NamespacedName: client.ObjectKey{
+						Name:      name,
+						Namespace: namespace,
+					},
+				},
+			}
+		})
+}
```

Comment:
wow super cool. 
I was at first scratching my head...I thought maybe if nodeclaim gets updated automatically other objects would get reconciled. But I realize now not. 

if you modify a nodeclaim it would only run reconcile on that object. What this is doing is also ensuring reconcile on workspace object! Cool!